packages/std/src/math/uint128.rs

```diff
@@ -258,15 +315,16 @@ impl Uint128 {
         if denominator == 0 {
             panic!("Denominator must not be zero");
         }
-        let val: u128 = (self.full_mul(numerator) / denominator)
+        (self.full_mul(numerator) / Uint256::from(denominator))
             .try_into()
-            .expect("multiplication overflow");
-        Uint128::from(val)
+            .expect("multiplication overflow")
     }
 
     /// Multiplies two u128 values without overflow.
-    fn full_mul(self, rhs: impl Into<u128>) -> U256 {
-        U256::from(self.u128()) * U256::from(rhs.into())
+    pub fn full_mul(self, rhs: impl Into<u128>) -> Uint256 {
```

Comment:
This new API is nice. Could you add a CHANGELOG entry and an example doc for it?